packages/api-fetch/src/utils/response.js

```diff
@@ -0,0 +1,70 @@
+/**
```

Comment:
The flow here could be simplified a lot and more JSDOC and inline comments should be added.

Let's add a note, but not block this PR so we can get it in the RC.